src/unittest/banded_global_aligner.cpp

```diff
@@ -0,0 +1,1406 @@
+//
+//  banded_global_aligner.cpp
+//  
+//
+//  Created by Jordan Eizenga on 7/26/16.
+//
+//
+
+#include <stdio.h>
+#include "catch.hpp"
+#include "gssw_aligner.hpp"
+#include "vg.hpp"
+#include "banded_global_aligner.hpp"
+
+using namespace google::protobuf;
+
+namespace vg {
+    namespace unittest {
+        
+        TEST_CASE( "Banded global alignment works when read matches exactly",
+                   "[alignment][banded][mapping]" ) {
+            
+            VG graph;
+            
+            QualAdjAligner aligner = QualAdjAligner();
+            
+            Node* n0 = graph.create_node("AGTG");
+            Node* n1 = graph.create_node("C");
+            Node* n2 = graph.create_node("A");
+            Node* n3 = graph.create_node("TGAAGT");
+            
+            graph.create_edge(n0, n1);
+            graph.create_edge(n0, n2);
+            graph.create_edge(n1, n3);
+            graph.create_edge(n2, n3);
+            
+            string read = string("AGTGCTGAAGT");
+            Alignment aln;
+            aln.set_sequence(read);
+            
+            int band_width = 1;
+            BandedGlobalAlignmentGraph banded_aligner = BandedGlobalAlignmentGraph(aln,
+                                                                                   graph.graph,
+                                                                                   band_width);
+            
+            
+            banded_aligner.align(aligner.score_matrix, aligner.nt_table, aligner.gap_open,
+                                 aligner.gap_extension);
+            
+            const Path& path = aln.path();
+            
+            // is a global alignment
```

Comment:
Maybe all these different categories of checks should be their own SECTIONs so if they fail you get better output?